chat/models.py

```diff
@@ -0,0 +1,27 @@
+from django.db import models
+from django.utils import timezone
+
+
+# class ChatManager(models.Manager):
+#     def get_group_chat(self, group_id):
+#         return self.filter(group_id=group_id).order_by('date_time')
+#
+#     def get_student_course_chat(self, student_course_id):
+#         return self.filter(student_course_id=student_course_id).order_by('date_time')
+
+
+class Message(models.Model):
+    message_id = models.BigAutoField(primary_key=True)
+    # sender_id = models.ForeignKey("User", on_delete=models.CASCADE)
+    # student_course_id = models.ForeignKey("Student_Course", on_delete=models.CASCADE)
+    # group_id = models.ForeignKey("Group", on_delete=models.CASCADE)
+    date_time = models.DateTimeField(default=timezone.now,  blank=False)
+    message = models.CharField(max_length=255,  blank=False)
+    # objects = ChatManager()
+
+    # def save(self, *args, **kwargs):
+    #     if self.student_course_id is None and self.group_id is None:
+    #         raise Exception("You can't leave both student_course_id and group_id fields as null")
```

Comment:
According to what I've seen the exception that needs to be raised is `ValidationError`

Reply:
Fixed :)
I read the documentation about the implementation of custom model validations and I moved the validation to the clean method.
The clean method is not invoked on save() or create() by default, so I overriden the save method of the model and invoked the full_clean().